lang/cs/Org.Apache.REEF.Tests/Functional/FaultTolerant/TestResubmitTask.cs

```diff
@@ -0,0 +1,263 @@
+
+                VerifyContextTaskMapping(value.Id, value.GetActiveContext().Value.Id);
+                Assert.Contains(TaskKilledByDriver, failedExeption);
+
+                SubmitTask(value.GetActiveContext().Value);
+            }
+
+            private void VerifyContextTaskMapping(string taskId, string contextId)
+            {
+                lock (_lock)
+                {
+                    string expectedContextId;
+                    _contextTaskMapping.TryGetValue(taskId, out expectedContextId);
+                    Assert.Equal(expectedContextId, contextId);
+                }
+            }
+
+            public void OnNext(IRunningTask value)
+            {
+                Logger.Log(Level.Info, "Task running: " + value.Id);
+                switch (value.Id)
+                {
+                    case TaskId + "1":
+                    case TaskId + "2":
+                        value.Dispose(Encoding.UTF8.GetBytes(KillTaskCommandFromDriver));
+                        break;
+                    case TaskId + "3":
+                    case TaskId + "4":
+                        value.Send(Encoding.UTF8.GetBytes(CompleteTaskCommandFromDriver));
+                        break;
+                    default: 
+                        throw new Exception("It should not be reached.");
+                }
+            }
+
+            public void OnCompleted()
+            {
+                throw new NotImplementedException();
+            }
+
+            public void OnError(Exception error)
+            {
+                throw new NotImplementedException();
+            }
+
+            private static IConfiguration GetTaskConfigurationForCloseTask(string taskId)
+            {
+                return TaskConfiguration.ConfigurationModule
+                    .Set(TaskConfiguration.Identifier, taskId)
+                    .Set(TaskConfiguration.Task, GenericType<ResubmitTask>.Class)
+                    .Set(TaskConfiguration.OnClose, GenericType<ResubmitTask>.Class)
+                    .Set(TaskConfiguration.OnMessage, GenericType<ResubmitTask>.Class)
+                    .Build();
+            }
+        }
+
+        private sealed class ResubmitTask : ITask, IDriverMessageHandler, IObserver<ICloseEvent>
+        {
+            private readonly CountdownEvent _suspendSignal = new CountdownEvent(1);
+
+            [Inject]
+            private ResubmitTask()
+            {
+            }
+
+            public byte[] Call(byte[] memento)
+            {
+                Logger.Log(Level.Info, "Hello in ResubmitTask");
+                _suspendSignal.Wait();
+                return null;
+            }
+
+            public void Dispose()
+            {
+                Logger.Log(Level.Info, "Task is disposed.");
+            }
+
+            public void OnNext(ICloseEvent value)
+            {
+                if (value.Value != null && value.Value.Value != null)
```

Comment:
If this is the right way to check, we might have to revisit the design of `ICloseEvent ` :)